src/walker.ts

```diff
@@ -252,16 +252,42 @@ export const findNode: FindNode = (obj, findFn, options = {}) => {
   return node
 }
 
+const chunkPath = (path: string[], separator: string) => {
+  let nestedPath: string[] = []
+  const chunkedPath = []
+  const addNestedPath = () => {
+    if (nestedPath.length) {
+      chunkedPath.push(nestedPath.join(separator))
+      nestedPath = []
+    }
+  }
+  for (const key of path) {
+    if (/[0-9]+/.test(key)) {
+      addNestedPath()
+      chunkedPath.push(key)
+    } else {
+      nestedPath.push(key)
+    }
+  }
+  addNestedPath()
+  return chunkedPath
+}
```

Comment:
Here's where some functions from futil come in handy

```javascript
const chunkPath = _.flow(
  F.chunkBy((group, x) => !_.isNumber(x) && !_.isNumber(_.last(group))),
  _.map(([x, ...group]) => (_.isNumber(x) ? x : _.join('.', [x, ...group]))),
)
```

Reply:
Yeah, I thought about using `chunkBy` but it was pretty easy to to write something specific here.